File: Nov2020/14Nov/FindTheString.java
/*
Problem Name: Find the String
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/multi-dimensional/practice-problems/algorithm/find-the-string-4014dec6/
*/
import java.util.*;
class FindTheString {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt(), m = in.nextInt();
            int mat[][] = new int[n][26];
            for(int i=0;i<n;i++){
                String s = in.next();
                for(char c: s.toCharArray())
                    mat[i][c-'a']++;
            }
            String s = in.next();
            int len = s.length(), cur = 0;
            boolean possible = true;
            for(int i=0;i<len;i++){
                char c = s.charAt(i);
                if(mat[cur][c-'a'] == 0){
                    possible = false;
                    break;
                }
                mat[cur][c-'a']--;
                cur = (cur + 1)%n;
            }
            if(possible)
                System.out.println("Yes");
            else
                System.out.println("No");
        }
    }
}

File: Nov2020/14Nov/FindTheString.cpp
/*
Problem Name: Find the String
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/multi-dimensional/practice-problems/algorithm/find-the-string-4014dec6/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int t;
    cin>>t;
    while(t--){
        int n,m,found=0;
        string s;
        cin>>n>>m;
        char mat[n][m];
        for(int i=0;i<n;++i){
            for(int j=0;j<m;++j){
                cin>>mat[i][j];
            }
        }
        cin>>s;

        for(int i=0;i<s.length();++i){
            found=0;
            for(int j=0;j<m;++j){
                if(s[i]==mat[i%n][j]){
                    found=1;
                    mat[i%n][j]=' ';
                    break;
                }
            }
            if(found!=1){
                cout<<"No"<<endl;
                break;
            }  
        }
        if(found==1)    cout<<"Yes"<<endl;
    }
    return 0;
}
